app/routes/download_results.py

```diff
@@ -0,0 +1,31 @@
+from flask import Response, json
+import redis
+from rq import Connection, Queue
+
+from app import app
+from config import Configuration
+
+
+config = Configuration()
+
+
+@app.route('/download_results/<string:job_id>', methods=['GET'])
+def download_results(job_id=None):
+    """Returns a JSON file with the results."""
+
+    # default output
+    result = {'Error': 'Result has expired'}
+
+    if job_id is not None:
+        redis_url = Configuration.REDIS_URL
+        redis_conn = redis.from_url(redis_url)
+        with Connection(redis_conn):
+            q = Queue(name=Configuration.QUEUE)
+            task = q.fetch_job(job_id)
+        # get the result as json
+        result = dict(task.result)
```

Comment:
Same here, you can check out different status codes if the result is not found.

Reply:
I have tried to implement this funcionality, but it seems to have no effect. Because 'Download results' and 'Download plot' buttons only appear when the classification is already done.
<img width="707" alt="Immagine 2022-03-30 215532" src="https://user-images.githubusercontent.com/73552095/160919960-1404453c-2dc2-4dc6-918a-b44cad648e32.png">
So It is not possible to press one button and see what happens if the task is in queue or when the task is waiting to be completed.
The only case that in my opinion can work is the case in which the task fails.
I have tried to implement the behavior with this code:
   if task.is_finished:
            # get the result as json
            result = dict(task.result)
            return Response(
                json.dumps(result),
                mimetype="text/json",
                headers={"Content-disposition": "attachment; filename=result.json"})
        elif task.is_queued:
            return Response( '102-task is in-queue', mimetype="text/html") , 102
        elif task.is_started:
            return Response('102-waiting for task complation', mimetype="text/html"), 102
        elif task.is_failed:
            return Response('503-task failed', mimetype="text/html"), 503

Reply:
Ok, looks good to me! So I would just deal with the task failed problem, for which I would not make the button appear at all. However, I will consider this issue resolved. Thank you for the clarification! :+1: